.github/workflows/ci.yaml

```diff
@@ -66,6 +83,8 @@ jobs:
   upstream-dev:
     name: Build (upstream-dev)
     runs-on: ubuntu-latest
+    needs: detect-ci-trigger
+    if: needs.detect-ci-trigger.outputs.triggered == 'false'
     steps:
     - uses: actions/checkout@v2
```

Comment:
I recommend using the following step to cancel previous runs

```yaml
      - name: Cancel previous runs
        uses: styfle/cancel-workflow-action@0.8.0
        with:
          access_token: ${{ github.token }}
```

This needs to be the first step in your workflow (so, this should be fore the `actions/checkout` step).

Reply:
Note: This step doesn't work for pushes on the main (default) branches. It works great on PRs though

Reply:
Nice! Thank you! This has to be added for each job right?

Reply:
Do i need to add the token to my secrets? Or is that a default somehow?

Reply:
`github.token` is the default and is available by default. So, you don't need a custom token

Reply:
> Nice! Thank you! This has to be added for each job right? 

I tend to use it on job that gets trigger on pull_requests events so as to avoid doing unnecessary work. I think putting it in the `build` job would suffice for the time being.

Reply:
I have put it into all of them for now. Especially the cloud tests spawn a lot of jobs (which caused this backlog). But now it seems to work much faster indeed!